pmlc/rt/level_zero/level_zero_utils.cc

```diff
@@ -0,0 +1,469 @@
+// Copyright 2020 Intel Corporation
+
+#include "pmlc/rt/level_zero/level_zero_utils.h"
+#include "pmlc/util/logging.h"
+
+namespace pmlc::rt::level_zero::lzu {
+
+#define MYASSERT(x)                                                            \
+  {                                                                            \
+    if (!(x)) {                                                                \
+      std::ostringstream oss;                                                  \
+      oss << "Failed in " << __func__ << " at " << __LINE__;                   \
+      throw std::runtime_error(oss.str());                                     \
+    }                                                                          \
+  }
+#define EXPECT_EQ(x, y) MYASSERT((x) == (y))
+#define EXPECT_NE(x, y) MYASSERT((x) != (y))
+#define EXPECT_GT(x, y) MYASSERT((x) > (y))
+#define EXPECT_TRUE(x) MYASSERT((x))
+#define ASSERT_EQ(x, y) EXPECT_EQ(x, y)
```

Comment:
These seem like the macros I would use in a GTest. I would rename these to avoid confusion.

Reply:
Thanks. Adds `PMLC_LEVEL_ZERO_` prefix to these macros.